### PR TITLE
Feat: code reduction using timeseriesexplorer.

### LINKED ISSUE #1557: Use TimeSeriesExplorer component in state pages.
**Is your feature request related to a problem? Please describe.**
Currently, home.js adds timeseries by adding the component `<TimeSeriesExplorer>` which contains the header and controls to operate the timeseries graphs. However, this is not the case in state pages, we're adding `<TimeSeries>` directly and rewriting the contents of `<TimeSeriesExplorer>` again over here.

**Describe the solution you'd like**
Use `<TimeSeriesExplorer>` in state pages as well and handling the input by passing relevant props. You can use css to disable the select dropdown from rendering (similar strategy to how map explorer is re-used in state pages).

**Describe alternatives you've considered**
Open to more suggestions on abstractions if someone can do it elegantly, ideally would want to reduce code duplication.

### PR DESCRIPTION
**Pull request by Binu42**
**Description of PR**
* this PR will Reduce code in `state.js` using `timeseriesexplorer.js` inspite  of `timeseries.js` and writting html for select mode and tabs.

**Relevant Issues**  
Fixes #1557 

**Checklist**

- [x] Compiles and passes lint tests
- [x] Properly formatted
- [x] Tested on desktop
- [x] Tested on phone

### CODE REVIEW

File: src/components/state.js [hunk 1]
```diff
@@ -384,69 +373,7 @@ function State(props) {
...
-                  </div>
-                </div>
-
-                <TimeSeries
-                  timeseries={timeseries}
-                  type={graphOption}
-                  mode={timeseriesMode}
-                  logMode={timeseriesLogMode}
-                />
+                <TimeSeriesExplorer stateTimeSeries={timeseries} />
```
Comment (shuklaayush) [inline src/components/state.js]: Best if you can use the existing timeseries variable.

File: src/components/timeseriesexplorer.js [hunk 2]
```diff
@@ -102,33 +103,39 @@ function TimeSeriesExplorer({
...
+        )}
       </div>
 
       <TimeSeries
-        timeseries={timeseries[activeStateCode]}
+        timeseries={
+          timeseries && activeStateCode
+            ? timeseries[activeStateCode]
+            : stateTimeSeries
+        }
```
Comment (shuklaayush) [inline src/components/timeseriesexplorer.js]: Might be better to instead pass only the state timeseries from both home.js and state.js. The complete timeseries data for all the states is not used anywhere in either TimeSeriesExplorer or TimeSeries. No need for the if conditions as well then.

File: src/components/timeseriesexplorer.js [hunk 3]
```diff
@@ -102,33 +103,39 @@ function TimeSeriesExplorer({
...
         stateCode={activeStateCode}
```
Comment (shuklaayush) [inline src/components/timeseriesexplorer.js]: I know it's not related to this PR, but can you remove stateCode from here and from inside TimeSeries as well? I don't think it's used anywhere.

**Review by shuklaayush — CHANGES_REQUESTED** [review]

**Review by shuklaayush — APPROVED** [review]
LGTM

**Comment by Binu42** [comment]
@shuklaayush i forgot to remove this `<div className="TimeSeriesExplorer">` from state.js can you please remove it or i should raise another PR for this otherwise graphs will not occupy full space in mobiles.

**Comment by shuklaayush** [comment]
@Binu42 No worries, just pushed a fix :)

**Pull request merged**